### ISSUE TITLE
Model name inconsistency in `ilab model list` and the related rest api

### ISSUE THREAD

**Issue by vishnoianil** [instructlab/instructlab#2763]
**What is the current experience of ilab you've found issue with?**
I have a `granite-7b-lab` model hosted through `ilab model serve`.  `ilab model list` lists the model name as `instructlab/granite-7b-lab`.

```text
 ilab model list
+----------------------------+---------------------+---------+
| Model Name                 | Last Modified       | Size    |
+----------------------------+---------------------+---------+
| instructlab/granite-7b-lab | 2024-12-04 22:09:20 | 12.6 GB |
+----------------------------+---------------------+---------+
```
But when i fetch the model list through the rest api, model name is the absolute directory path to the model file.

```
curl -X GET http://localhost:8000/v1/models
{"object":"list","data":[{"id":"/home/ec2-user/.cache/instructlab/models/instructlab/granite-7b-lab","object":"model","created":1733531472,"owned_by":"vllm","root":"/home/ec2-user/.cache/instructlab/models/instructlab/granite-7b-lab","parent":null,"max_model_len":4096,"permission":[{"id":"modelperm-69ef8f2c3476404b8a97361f74d255ce","object":"model_permission","created":1733531472,"allow_create_engine":false,"allow_sampling":true,"allow_logprobs":true,"allow_search_indices":false,"allow_view":true,"allow_fine_tuning":false,"organization":"*","group":null,"is_blocking":false}]}]}
```

And API to chat with the model requires the model name with full path, otherwise it throws 404.

**How do you think the described experience should be?**
I think a better user experience would be to have a consistency across the output of `ilab model list` and the related rest API. Providing the absolute path to the model seems unnecessary and might not be ideal (exposing the username in the path). 

**What is the benefit of the change you are proposing?**

API should provide the same model name as `ilab model list`, and chat API's should be able to accept it as well.

**Comment by cdoern** [comment]
I don't think we directly have control over what is returned by those requests, as what you are receiving is the llama-cpp or vllm server. 

ilab does not currently have its own REST API, but one will be necessary for consumers of ilab outside of the CLI.

**Comment by jpodivin** [comment]
I don't have a full answer to that problem. But at very least we can display path to the model in the list to make writing requests easier.

https://github.com/instructlab/instructlab/pull/2941

**Comment by ktam3** [comment]
@vishnoianil - could you follow up on what Charlie asked?

**Comment by jpodivin** [comment]
I have a PR now for the usage of model names in REST API calls. https://github.com/instructlab/instructlab/pull/3080
The idea is that you can now refer to the model using both the absolute path and the shorter name.

**Comment by vishnoianil** [comment]
> I don't have a full answer to that problem. But at very least we can display path to the model in the list to make writing requests easier.
> 
> [#2941](https://github.com/instructlab/instructlab/pull/2941)

That makes sense to me, at least the model name is consistent at both the place. 

In my opinion we should have it consistent because i believe that's a better user experience. For example, i used `ilab model list` to figure out the model name, and than i used the same model name with `chat/completions` API and it failed, because it expect the full path to the model. So i had to than pull the model list using the rest api and that returns the full path. Not a big deal to debug this, but i believe this increases ilab cli's usability.
